Lib/test/support/__init__.py

```diff
@@ -856,6 +856,20 @@ def check_cflags_pgo():
     return any(option in cflags_nodist for option in pgo_options)
 
 
+def check_bolt_optimized():
+    # Always return false, if the platform is WASI.
+    if is_wasi:
+        return False
+    # BOLTed binary can be checked based on ELF information.
+    # link: https://github.com/llvm/llvm-project/issues/60253
+    binary_path = sys.executable
+    with open(binary_path, 'rb') as f:
+        binary = f.read()
+        if b'.note.bolt_info' in binary:
```

Comment:
Can you call `readelf -e <binary_path>` and grep `.note.bolt_info`? Checking headers is efficient and robust.

Reply:
I also think it will be the best way to handle this issue. But I am not sure it will be okay to call external binary for this. I may need to get opinions from @vstinner and @erlend-aasland WDYT?

Reply:
Why not checking sysconfig? This check looks slow and not reliable.

Reply:
> Why not checking sysconfig? This check looks slow and not reliable.

Do you have any recommendation points? CFLAG does not guarantee the build is BOLTed.

Reply:
Maybe test:

```
'--enable-bolt' in sysconfig.get_config_var('CONFIG_ARGS')
```

Reply:
Same, it does not gurentee.

Reply:
You cannot expect "readelf" to be available, it would be a new dependency. I dislike looking for a string in a binary without parsing the ELF binary.

> Same, it does not gurentee.

Skipping the test if the binary was not optimized by BOLT is fine with me.

Reply:
Only put the code in test_gdb if you're worried that the the test is not reliable. And add a comment to explain it.

Reply:
Well than no way to display the optimization information but okay got it.

Reply:
> You cannot expect "readelf" to be available, it would be a new dependency. I dislike looking for a string in a binary without parsing the ELF binary.

We call external binaries in `test.support` to determine macOS features, so there is precedent for doing that sort of thing; is `readelf` a standalone package, or is it part of the compiler toolchain?

Reply:
> readelf

AFAIK, it's part of binutils: https://sourceware.org/binutils/docs/binutils/readelf.html

Reply:
> AFAIK, it's part of binutils: https://sourceware.org/binutils/docs/binutils/readelf.html

Thanks. `binutils` is preinstalled on the Ubuntu runners GitHub provides, so there is no more dependencies for the CI there. Also, anyone who wants to build their own CPython probably already did `apt install build-essential`, which includes `binutils`. Looks to me like there are no more deps introduced. What about macOS?